integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/search/loading/SearchQueryEntityLoadingGraphIT.java

```diff
@@ -70,25 +63,34 @@ protected BackendMock backendMock() {
 
 	@Override
 	protected SessionFactory sessionFactory() {
-		return setupHolder.sessionFactory();
+		return sessionFactory;
 	}
 
-	@ReusableOrmSetupHolder.SetupParams
-	public List<?> setupParams() {
-		return Arrays.asList( mapping, model );
+	@Override
+	protected SingleTypeLoadingModel<T> model() {
+		return model;
 	}
 
-	@ReusableOrmSetupHolder.Setup
-	public void setup(OrmSetupHelper.SetupContext setupContext, ReusableOrmSetupHolder.DataClearConfig dataClearConfig) {
-		backendMock.expectAnySchema( model.getIndexName() );
-		setupContext.withConfiguration( c -> mapping.configure( c, model ) );
+	@Override
+	protected SingleTypeLoadingMapping mapping() {
+		return mapping;
+	}
 
-		dataClearConfig.preClear( model.getIndexedClass(), model::clearContainedEager );
-		dataClearConfig.clearOrder( model.getContainedClass(), model.getIndexedClass() );
+	@ParameterizedSetup
+	@MethodSource("params")
+	void setup(SingleTypeLoadingModel<T> model, SingleTypeLoadingMapping mapping) {
+		this.model = model;
+		this.mapping = mapping;
+		backendMock.expectAnySchema( model.getIndexName() );
+		sessionFactory = ormSetupHelper.start().withConfiguration( c -> mapping.configure( c, model ) )
+				.dataClearing( true, config -> config.clearIndexData( false )
+						.preClear( model.getIndexedClass(), model::clearContainedEager )
+						.clearOrder( model.getContainedClass(), model.getIndexedClass() ) )
+				.setup();
```

Comment:
I just tried to run this test, and it appears that session factories all get closed at the very end of the test run, instead of being closed when moving from one setup to the next. This could cause serious problems. Could you please have a look?

To reproduce, put a breakpoint on this line as well as in `org.hibernate.internal.SessionFactoryImpl#close`, then run the test in debug mode.